addons/binding/org.openhab.binding.amazonechocontrol/src/main/java/org/openhab/binding/amazonechocontrol/internal/Connection.java

```diff
@@ -0,0 +1,925 @@
+                    "{\"bluetoothDeviceAddress\":\"" + address + "\"}", true, true, null);
+        }
+    }
+
+    public void playRadio(Device device, @Nullable String stationId) throws IOException, URISyntaxException {
+        if (StringUtils.isEmpty(stationId)) {
+            command(device, "{\"type\":\"PauseCommand\"}");
+        } else {
+            makeRequest("POST",
+                    alexaServer + "/api/tunein/queue-and-play?deviceSerialNumber=" + device.serialNumber
+                            + "&deviceType=" + device.deviceType + "&guideId=" + stationId
+                            + "&contentType=station&callSign=&mediaOwnerCustomerId=" + device.deviceOwnerCustomerId,
+                    "", true, true, null);
+        }
+    }
+
+    public void playAmazonMusicTrack(Device device, @Nullable String trackId) throws IOException, URISyntaxException {
+        if (StringUtils.isEmpty(trackId)) {
+            command(device, "{\"type\":\"PauseCommand\"}");
+        } else {
+            String command = "{\"trackId\":\"" + trackId + "\",\"playQueuePrime\":true}";
+            makeRequest("POST",
+                    alexaServer + "/api/cloudplayer/queue-and-play?deviceSerialNumber=" + device.serialNumber
+                            + "&deviceType=" + device.deviceType + "&mediaOwnerCustomerId="
+                            + device.deviceOwnerCustomerId + "&shuffle=false",
+                    command, true, true, null);
+        }
+    }
+
+    public void playAmazonMusicPlayList(Device device, @Nullable String playListId)
+            throws IOException, URISyntaxException {
+        if (StringUtils.isEmpty(playListId)) {
+            command(device, "{\"type\":\"PauseCommand\"}");
+        } else {
+            String command = "{\"playlistId\":\"" + playListId + "\",\"playQueuePrime\":true}";
+            makeRequest("POST",
+                    alexaServer + "/api/cloudplayer/queue-and-play?deviceSerialNumber=" + device.serialNumber
+                            + "&deviceType=" + device.deviceType + "&mediaOwnerCustomerId="
+                            + device.deviceOwnerCustomerId + "&shuffle=false",
+                    command, true, true, null);
+        }
+    }
+
+    public void textToSpeech(Device device, String text) throws IOException, URISyntaxException {
+        Map<String, Object> parameters = new Hashtable<String, Object>();
+        parameters.put("textToSpeak", text);
+        executeSequenceCommand(device, "Alexa.Speak", parameters);
+    }
+
+    // commands: Alexa.Weather.Play, Alexa.Traffic.Play, Alexa.FlashBriefing.Play, Alexa.GoodMorning.Play,
+    // Alexa.SingASong.Play, Alexa.TellStory.Play, Alexa.Speak (textToSpeach)
+    public void executeSequenceCommand(Device device, String command, @Nullable Map<String, Object> parameters)
+            throws IOException, URISyntaxException {
+        Gson gson = new Gson();
```

Comment:
This is potentially expensive - you should instantiate it only once as a field in that class.